api/types/provisioning.go

```diff
@@ -376,3 +392,21 @@ func (p ProvisionTokenV1) String() string {
 	return fmt.Sprintf("ProvisionToken(Roles=%v, Expires=%v)",
 		p.Roles, expires)
 }
+
+func (a *ProvisionTokenSpecV2GitHub) checkAndSetDefaults() error {
+	if len(a.Allow) == 0 {
+		return trace.BadParameter("the %q join method requires defined token allow rules", JoinMethodGitHub)
+	}
+	for _, rule := range a.Allow {
+		repoSet := rule.Repository != ""
+		ownerSet := rule.RepositoryOwner != ""
+		subSet := rule.Sub != ""
+		if !(subSet || ownerSet || repoSet) {
+			return trace.BadParameter(
+				`allow rule for %q must abide by secure guidelines. check documentation.`,
```

Comment:
Can we create a more actionable error message here that tells the user what is wrong and how to fix it? To me, this kind of sounds like "you messed up, better go check the docs" which feels like it could be frustrating rather than helpful.